### PR TITLE
NRF albino penguin

### PR DESCRIPTION
**Pull request by ghost**
This adds the giant albino penguin from Lovecraftian lore to the game.

### CODE REVIEW

File: data/json/monstergroups.json [hunk 1]
```diff
@@ -603,7 +604,8 @@
     "monsters" : [
       { "monster" : "mon_sewer_fish", "freq" : 300, "cost_multiplier" : 0, "pack_size" : [2,6] },
       { "monster" : "mon_sewer_snake", "freq" : 240, "cost_multiplier" : 0 },
-      { "monster" : "mon_sewer_rat", "freq" : 100, "cost_multiplier" : 0, "pack_size" : [3,12] }
+      { "monster" : "mon_sewer_rat", "freq" : 100, "cost_multiplier" : 0, "pack_size" : [3,12] },
+	  { "monster" : "mon_albino_penguin", "freq" : 10, "cost_multiplier" : 5, "pack_size" : [3,9] }
```
Comment (Coolthulhu) [inline data/json/monstergroups.json]: Tabs not allowed, those have to be 4 spaces instead.

Comment (nrfowler) [inline data/json/monstergroups.json]: thanks, have added this to the second commit

**Comment by chaosvolt** [comment]
Tekeli-li.

File: data/json/monsters.json [hunk 2]
```diff
@@ -5040,6 +5040,36 @@
...
+    "melee_cut":8,
+    "dodge":2,
+    "armor_bash":6,
+    "armor_cut":2,
+    "vision_day":5,
+	"vision_night":5,
+    "hp":200,
+    "death_function":"NORMAL",
+	"special_attacks":[["BITE", 15],["IMPALE",20]],
+    "description":"Eight-feet-tall penguins bred by the Elder Things. Placid unless otherwise provoked.",
```
Comment (Rivet-the-Zombie) [inline data/json/monsters.json]: Double spacing after periods, please.

**Comment by vache** [comment]
I would remove the references to the elder things in the description.  Most of the other nether monsters  it just includes characteristics in the description and doesn't reference the "source" of the creature.  Also, as a general rule, I'm not sure we have full Lovecraftian mythos going on, just some creatures we've lifted from the lore.

**Comment by chaosvolt** [comment]
Agreed. We can leave the reference for the player to figure out, in most cases.

File: data/json/monsters.json [hunk 3]
```diff
@@ -5040,6 +5040,36 @@
...
+    "speed":50,
+    "melee_skill":7,
+    "melee_dice":4,
+    "melee_dice_sides":6,
+    "melee_cut":8,
+    "dodge":2,
+    "armor_bash":6,
+    "armor_cut":2,
+    "vision_day":5,
+	"vision_night":5,
```
Comment (Coolthulhu) [inline data/json/monsters.json]: Tabs
Need to be 4 spaces instead

Any line that looks "weird" in github while looking OK in text editor has tabs.

File: data/json/monstergroups.json [hunk 4]
```diff
@@ -453,7 +453,8 @@
       { "monster" : "mon_bat", "freq" : 600, "cost_multiplier" : 5, "pack_size" : [6,32] },
       { "monster" : "mon_bear", "freq" : 100, "cost_multiplier" : 10, "pack_size" : [1,3] },
       { "monster" : "mon_cougar", "freq" : 100, "cost_multiplier" : 20, "pack_size" : [1,2] },
-      { "monster" : "mon_bear_smoky", "freq" :  100, "cost_multiplier" : 0, "starts" : 672 }
+      { "monster" : "mon_bear_smoky", "freq" :  100, "cost_multiplier" : 0, "starts" : 672 },
+      { "monster" : "mon_albino_penguin", "freq" : 10, "cost_multiplier" : 25, "pack_size" : [3,9] }
```
Comment (Coolthulhu) [inline data/json/monstergroups.json]: 9 in a cave? Caves are very small at the moment.

**Comment by chaosvolt** [comment]
I've seen craploads of rats and bats in a cave and yeah, it can be hectic. Those are tiny though, whereas these big enough to wrestle a shoggoth.

Speaking of which, I forgot how big the penguins where according to In The Mountains of Madness, but fairly certain they weren't that big. o.O

File: data/json/monsters.json [hunk 5]
```diff
@@ -5040,6 +5040,36 @@
...
     "flags":["SEES", "SMELLS", "SWIMS", "PLASTIC", "SLUDGEPROOF", "ACID_BLOOD", "ACID_PROOF", "NOHEAD", "REGENERATES_50", "ABSORBS", "REGEN_MORALE", "NOGIB"]
     },{
+    "type":"MONSTER",
+    "id":"mon_albino_penguin",
+    "name": "albino penguin",
+    "species":"NETHER",
+    "default_faction":"nether",
+    "symbol":"V",
+    "color":"white",
+    "size":"HUGE",
```
Comment (Coolthulhu) [inline data/json/monsters.json]: Size large would be more fitting. At 8 feet tall, they should have mass more comparable to that of a cow than of a tank.

**Comment by nrfowler** [comment]
That's true, I wanted this penguin to be more impressive than a cow, but in fact a cow placed vertically could be up to 8.5 feet tall nose-to-tail.

**Comment by nrfowler** [comment]
On the contrary it looks like 9 penguins would fit in the lower level of caves as caves are 20x20, but I'm going to remove them from caves for other reasons. A cow sized animal that is inherently unstable couldn't navigate in and out of a cave (at least the majority of caves), and it must do so to feed on fish in a nearby body of water. Penguins could easily enter sewers and there are actually fish there.

**Comment by chaosvolt** [comment]
Might still suggest having them show up in much smaller numbers in caves. Or if I knew any other logical locations for them...

**Comment by nrfowler** [comment]
Having reviewed the locations again, I plan on adding them to the ice lab group - which is where they would have to be coming from.

**Pull request merged**